generator/app.go

```diff
@@ -294,6 +300,21 @@ func (c *Context) Validate() []error {
 					errors = append(errors, fmt.Errorf("%s: has no subprojects", group.Dir))
 				}
 			}
+			if prefix != "committee" && prefix != "sig" {
+				if len(group.Subprojects) > 0 {
+					errors = append(errors, fmt.Errorf("%s: only sigs and committees can own code / have subprojects, found: %v", group.Dir, group.Subprojects))
+				}
+			}
+			for _, subproject := range group.Subprojects {
+				if len(subproject.Owners) == 0 {
+					errors = append(errors, fmt.Errorf("%s/%s: subproject has no owners", group.Dir, subproject.Name))
+				}
+				for _, ownerURL := range subproject.Owners {
+					if !reRawGitHubURL.MatchString(ownerURL) && !reGitHubURL.MatchString(ownerURL) {
```

Comment:
Why do we allow both rawgithuburl and githuburl links? Would it be better to restrict to just rawgithub ones to ensure that the OWNERS files in sigs.yaml can be programmatically read?

Reply:
I was thinking of trending in the direction of being flexible in what we parse, and maybe eventually switching from raw to something else.  If we were to (progressively?) switch from raw to github urls, `githubURL` would pass them through unchanged, and `orgRepoPath` can parse them.

Higher-level, break out the types+parsing+validation to be more reusable as a lib for automation (eg: validating sigs.yaml external assets https://github.com/kubernetes/community/issues/5425)

The original motivation for raw links was: automation can automatically download file contents.  However, I think generating raw links is more of a pain to contributors.

If such automation exists in the wild today, I would love to hear about it, and we can talk about how not to break it.  I know I built some in the past, but at this point I would rather rewrite it (or help someone else do so) under https://github.com/kubernetes/community/issues/5425